qiskit_nature/transformers/second_quantization/electronic/seniority_zero_transformer.py

```diff
@@ -0,0 +1,242 @@
+            )
+
+        particle_number = grouped_property.get_property(ParticleNumber)
+        if particle_number is None:
+            raise QiskitNatureError(
+                "The provided `GroupedElectronicProperty` does not contain a `ParticleNumber` "
+                "property, which is required by this transformer!"
+            )
+        particle_number = cast(ParticleNumber, particle_number)
+
+        self._num_modes = particle_number.num_spin_orbitals // 2
+
+        grouped_property_transformed = self._transform_property(grouped_property)
+
+        return grouped_property_transformed  # type: ignore[return-value]
+
+    def _transform_property(self, prop: Property) -> Property:
+        """Transforms a `Property` object.
+
+        This is a recursive reduction, iterating `GroupedProperty` objects when encountering one.
+
+        Args:
+            property: the property object to transform.
+
+        Returns:
+            The transformed property object.
+
+        Raises:
+            TypeError: if an unexpected Property subtype is encountered.
+        """
+        transformed_property: Property
+        # Code for recursion is copied from ActivateSpaceTransformer
+        if isinstance(prop, GroupedProperty):
+            transformed_property = deepcopy(prop)
+
+            for internal_property in iter(prop):
+                try:
+                    transformed_internal_property = self._transform_property(internal_property)
+                    if transformed_internal_property is not None:
+                        transformed_property.add_property(transformed_internal_property)
+                    else:
+                        transformed_property.remove_property(internal_property)
+                except TypeError:
+                    logger.warning(
+                        "The Property %s of type %s could not be transformed!",
+                        internal_property.name,
+                        type(internal_property),
+                    )
+                    continue
+
+            if len(transformed_property._properties) == 0:
+                transformed_property = None
+
+        elif isinstance(prop, ElectronicEnergy):
+            h_1 = prop.get_electronic_integral(ElectronicBasis.MO, 1).get_matrix()
+            h_2 = prop.get_electronic_integral(ElectronicBasis.MO, 2).get_matrix()
+
+            hr_1, hr_2 = self._restricted_electronic_integrals(h_1, h_2)
+
+            transformed_property = ElectronicEnergy.from_raw_integrals(
+                ElectronicBasis.MO, hr_1, hr_2
+            )
+            transformed_property._shift = prop._shift.copy()
+            transformed_property.nuclear_repulsion_energy = prop.nuclear_repulsion_energy
+            transformed_property.reference_energy = prop.reference_energy
+
+            transformed_property.second_q_ops = partial(  # type: ignore[assignment]
+                second_q_ops_electronic_energy, transformed_property
+            )
```

Comment:
This monkeypatching approach is currently a bit of a hack. An alternative to this approach is the use of a Strategy pattern as was already mentioned in the PR description. While this would make things slightly cleaner, it would have the downside of extracting all current `Property.second_q_ops()` implementations into a single "default" strategy.
A transformer which would affect this logic would then ship its own strategy which needs to overwrite the methods for all supported property types.

In the future, adding a new property will then cause all existing strategies to be extended by another implementation. Thus we have a NxM problem with N being the number of properties and M the number of strategies.

I am open to better suggestions of handling this :+1: